### ISSUE TITLE
Wrong warning "Code is unreachable"

### ISSUE THREAD

**Issue by kaldown** [fannheyward/coc-pyright#341]
https://github.com/microsoft/pyright/issues/1532

For some reason `Pyright` thinks this code is unreachable - this is clearly a mistake.
```
from contextlib import suppress

seconds = 0
with suppress(ValueError):
    seconds = int('a')

if not seconds:
    seconds = 42
```

Then it thinks all the other code, out of scope is broken too. Which is another (maybe consecutive) misconception.

**Comment by yaegassy** [comment]
I can't reproduce it. There is no diagnostic message from pyright.

![coc-pyright-issue-341-1](https://user-images.githubusercontent.com/188642/109081206-42142f80-7745-11eb-90bc-f86d287cad41.gif)

**Comment by fannheyward** [comment]
Upgrade your coc-pyright and try again.

**Comment by kaldown** [comment]
I'm on this commit `9edbfb3162abde27e260b4b97bd4b616d2f5182c` of coc-pyright.

![2021-02-25_550x236_scrot](https://user-images.githubusercontent.com/3754017/109143220-d47df880-7770-11eb-9236-da419c0763e0.png)

**Comment by yaegassy** [comment]
The screenshot you attached appears to be different from the Python code you reported in the original pyright and coc-pyright issues.

If you want to hide the "Code is unreachable", set `"diagnostic.showUnused": false` in `coc-settings.json`.

**Comment by kaldown** [comment]
> The screenshot you attached looks to me to be different from the Python code you reported in the original pyright and coc-pyright issues.
> 
> If you want to hide the "Code is unreachable", set `"diagnostic.showUnused": false` in `coc-settings.json`.

This idea is the same. Pyright doesn't understand contextmanager.

Why would I disable unused variables. What the point using pyright then.

I only can guess that unused is the same as unreachable. Because I reported problem with unreachable, not unused

**Comment by kaldown** [comment]
![2021-03-02_456x1036_scrot](https://user-images.githubusercontent.com/3754017/109687032-ebb74e80-7b93-11eb-8056-b42480ba4b7c.png)

this is so annoying. How to fix that without disabling **pyright** features, suchs as `diagnostic.showUnused`?

It is clearly reachable as you can see on an example. I created 2 variables, with and without affecting mutation within `suppress` ctx

**Comment by yaegassy** [comment]
It is a diagnostic message at the "hint level" (`[Pyright] [H] ...`) . It is not pointing out an error that the code is not executable.

**Comment by kaldown** [comment]
Sure, but imagine having hundreds of lines down, and all of them being marked as unreached.

Disabling diagnostics is not a solution, since I need those to be showed up.

Having only 4 plugins: coc, coc-pyright, undotree, fzf.

With the at most default configs.
How is that not reproducable. Or what to do to understand why it marked completelly okay code?

Will try to enable features one by one to find out why it happening.

**Comment by yaegassy** [comment]
This is not a coc-pyright issue.  You may want to suggest it in an issue of pyright itself or on gitter.

**Comment by fannheyward** [comment]
Tested with this code, in a new venv, can't reproduce.

```python
#!/usr/bin/env python3

from contextlib import suppress

seconds = 0
with suppress(ValueError):
    seconds = int(42)

if not seconds:
    seconds = 3

print(seconds)

## NEW
seconds_new = 0
with suppress(ValueError):
    seconds_new = int("a")

if not seconds_new:
    seconds_new = 5

print(seconds_new)
```

**Comment by kaldown** [comment]
> This is not a problem with coc-pyright or any other Vim plugin.  You may want to suggest it in an issue of pyright itself or on gitter.
> 
> Or it's a problem with your python environment or your code. (I can't reproduce it)

I did send it to ms/pyright from the beginning. See link at the start.
He said it is yours.

What is gitter?

**Comment by fannheyward** [comment]
@kaldown you can get your coc-pyright version from `:CocList extensions`.

**Comment by yaegassy** [comment]
@kaldown Unfortunately, problems that cannot be reproduced cannot be solved. :(

First, try updating to the latest version of coc-pyright. Currently, the latest version is `v1.1.117`.

I looked at your dotfile. https://github.com/kaldown/dotfiles/blob/master/.config/nvim/plug_config/base.vim#L64

Is your coc-pyright installed correctly? (Did you build the new version correctly?)

If you are installing with "vim-plug", it is useful to set `"Plug 'fannheyward/coc-pyright', {'do': 'yarn install --frozen-lockfile'}`.

It is more useful if you use `:CocInstall coc-pyright`.

**Comment by kaldown** [comment]
> @kaldown Unfortunately, problems that cannot be reproduced cannot be solved. :(
> 
> First, try updating to the latest version of coc-pyright. Currently, the latest version is `v1.1.117`.
> 
> I looked at your dotfile. https://github.com/kaldown/dotfiles/blob/master/.config/nvim/plug_config/base.vim#L64
> 
> Is your coc-pyright installed correctly? (Did you build the new version correctly?)
> 
> If you are installing with "vim-plug", it is useful to set `"Plug 'fannheyward/coc-pyright', {'do': 'yarn install --frozen-lockfile'}`.
> 
> It is more useful if you use `:CocInstall coc-pyright`.

Thank you.
Initially I installed `coc-pyright` with `CoC marketplace`, only then added in through `vim-plug`. Could be ther are overriden.

`coc-pyright 1.1.101 ~/.config/coc/extensions/node_modules/coc-pyright`.


Installed via `:CocInstall coc-pyright` and it worked!

Thank you very much @yaegassy .

Wonder why is that so. With vim-plug I have the most recent master commit.
Perhaps there are two versions, from `CoC`, and from `vim-plug`.

**Comment by fannheyward** [comment]
> https://github.com/kaldown/dotfiles/blob/master/.config/nvim/plug_config/base.vim#L64

You're using the _wrong_ way to install coc-pyright with vim-plug, check https://github.com/neoclide/coc.nvim/wiki/Using-coc-extensions#use-vims-plugin-manager-for-coc-extension

**Comment by yaegassy** [comment]
I found a similar issue on pyright itself. https://github.com/microsoft/pyright/issues/1376

I forced the pyright version of coc-pyright to `v1.1.101`, and the "bug" was reproduced.

<img width="458" alt="pyright-v1-1-101-bug" src="https://user-images.githubusercontent.com/188642/109780224-24196400-7c4a-11eb-9c79-b0c43eb7aab2.png">

**Comment by kaldown** [comment]
> > https://github.com/kaldown/dotfiles/blob/master/.config/nvim/plug_config/base.vim#L64
> 
> You're using the _wrong_ way to install coc-pyright with vim-plug, check https://github.com/neoclide/coc.nvim/wiki/Using-coc-extensions#use-vims-plugin-manager-for-coc-extension

Thanks, I already fixed that as @yaegassy suggested
